src/tests/ftest/slurm_setup.py

```diff
@@ -0,0 +1,281 @@
+
+    Args:
+        args (Namespace): Commandline arguments
+
+    """
+    # Check if file exists on slurm control node
+    if execute_cluster_cmds(args.control, ["ls /etc/munge/munge.key"]) > 0:
+        # Create one key on control node and then copy it to all slurm nodes
+        if execute_cluster_cmds(
+                    args.control, ["create-munge-key"], args.sudo) > 0:
+            exit(1)
+
+    # copy key to all nodes FROM slurmctl node;
+    cmd_list = [
+        "clush --copy -p -w {} /etc/munge/munge.key".format(args.nodes)]
+    if execute_cluster_cmds(args.control, cmd_list, args.sudo) > 0:
+        exit(1)
+
+    # Start Munge service on all nodes
+    all_nodes = NodeSet("{},{}".format(str(args.control), str(args.nodes)))
+    if execute_cluster_cmds(all_nodes, munge_startup, args.sudo) > 0:
+        exit(1)
+
+
+def start_slurm(args):
+    """Start the slurm services on all nodes.
+
+    Args:
+        args (Namespace): Commandline arguments
+
+    """
+    # Setting up slurm on all nodes
+    all_nodes = NodeSet("{},{}".format(str(args.control), str(args.nodes)))
+    cmd_list = [
+        "mkdir -p {0};chown {1}. {0}".format("/var/log/slurm", args.user),
+        "mkdir -p {0}d;mkdir -p {0}ctld;chown {1}. {0}ctld".format(
+            "/var/spool/slurm/", args.user)
+        ]
+    if execute_cluster_cmds(all_nodes, cmd_list, args.sudo) > 0:
+        exit(1)
+
+    # Startup the slurm control service
+    if execute_cluster_cmds(args.control, slurmctl_startup, args.sudo) > 0:
+        exit(1)
+    else:
+        # Startup the slurm service
+        if execute_cluster_cmds(all_nodes, slurm_srv_startup, args.sudo) > 0:
+            exit(1)
+
+
+def main():
+    """Set up test env with slurm."""
+    logging.basicConfig(
+        format="%(asctime)s %(levelname)-5s %(message)s",
+        datefmt=r"%Y/%m/%d %I:%M:%S",
+        name="slurm_setup", level=logging.DEBUG)
+
+    parser = argparse.ArgumentParser(prog="slurm_setup.py")
+
+    parser.add_argument(
+        "-n", "--nodes",
+        default=None,
+        help="Comma separated list of nodes to install slurm")
+    parser.add_argument(
+        "-c", "--control",
+        default=socket.gethostname().split('.', 1)[0],
+        help="slurm control node; test control node if None")
+    parser.add_argument(
+        "-u", "--user",
+        default=pwd.getpwuid(os.geteuid()).pw_name,
+        help="slurm user for config file; if none the current user is used")
+    parser.add_argument(
+        "-i", "--install",
+        action="store_true",
+        help="Install all the slurm/munge packages")
+    parser.add_argument(
+        "-r", "--remove",
+        action="store_true",
+        help="Install all the slurm/munge packages")
+    parser.add_argument(
+        "-s", "--sudo",
+        action="store_true",
+        help="Run all commands with privileges")
+
+    args = parser.parse_args()
+    logging.info("Arguments: {}".format(args))
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments